snowshu/adapters/base_sql_adapter.py

```diff
@@ -238,15 +239,18 @@ def quoted(val: str) -> str:
         raise NotImplementedError()
 
     def quoted_dot_notation(self, rel: Relation) -> str:
-        case_corrected_database = self._correct_case(rel.database)
-        case_corrected_schema = self._correct_case(rel.schema)
-        case_corrected_name = self._correct_case(rel.name)
+        case_corrected_database = self._correct_case(rel.database, 'database')
+        case_corrected_schema = self._correct_case(rel.schema, 'schema')
+        case_corrected_name = self._correct_case(rel.name, 'relation')
         return '.'.join([self.quoted(val) for val in 
                         (case_corrected_database, 
                          case_corrected_schema, 
                          case_corrected_name)])
 
-    def _correct_case(self, val: str) -> str:
+    def _correct_case(self, val: str, attr: str = None) -> str:  # noqa pylint: disable=too-many-function-args
         """The base case correction method for a sql adapter.
         """
-        return val if self.preserve_case else correct_case(val, self.DEFAULT_CASE == 'upper')
+        preserve_case = self.preserve_case[attr] if attr in ['database', 'schema', 'relation'] else False
```

Comment:
We should limit the values passed to this function to be those in this list. Currently, any arbitrary string can be passed in and assumed to be `False`